tests/Unit/HtmlProcessor/CssVariableEvaluatorTest.php

```diff
@@ -0,0 +1,394 @@
+                'expect' => '<p style="font-size: calc(.8rem * 1.2);"',
+            ],
+            'fallback value provided for undefined variable' => [
+                'css' => '
+                    p {
+                        color: var(--text-color, green);
+                    }
+                ',
+                'expect' => '<p style="color: green;">',
+            ],
+            'fallback value provided for defined variable' => [
+                'css' => '
+                    :root {
+                        --text-color: green;
+                    }
+                    p {
+                        color: var(--text-color, red);
+                    }
+                ',
+                'expect' => '<p style="color: green;">',
+            ],
+            'fallback value refererencing another variable which is defined' => [
+                'css' => '
+                    :root {
+                        --default-text-color: green;
+                    }
+                    p {
+                        color: var(--text-color, var(--default-text-color));
+                    }
+                ',
+                'expect' => '<p style="color: green;">',
+            ],
+            'fallback value refererencing another variable which is also not defined' => [
+                'css' => '
+                    p {
+                        color: var(--text-color, var(--default-text-color));
+                    }
+                ',
+                // The expected behaviour here is somewhat ambiguous.
+                'expect' => '<p style="color: var(--default-text-color);">',
+            ],
+            'nested fallback value' => [
+                'css' => '
+                    p {
+                        color: var(--text-color, var(--default-text-color, green));
+                    }
+                ',
+                'expect' => '<p style="color: green;">',
+            ],
+            'fallback value with parentheses' => [
+                'css' => '
+                    body {
+                        width: var(--page-width, calc(100vw - 20px));
+                    }
+                ',
+                'expect' => '<body style="width: calc(100vw - 20px);">',
+            ],
+            'fallback value with nested parentheses' => [
+                'css' => '
+                    body {
+                        width: var(--page-width, calc(100vw - 2 * calc(1rem + 10px)));
+                    }
+                ',
+                'expect' => '<body style="width: calc(100vw - 2 * calc(1rem + 10px));">',
+            ],
+            // Processing through `CssInliner` may result in the quotes being changed.
+            // A direct test with HTML is also performed for the following four scenarios.
+            'fallback value with single-quoted string containing opening parenthesis' => [
+                'css' => '
```

Comment:
I propose we switch to double quotes for those strings in PHP that contain (only) single quotes. Then we don't need to escape the single quotes.

Reply:
All of the strings contain both double and single quotes (one type for the `style` attribute value, and the other type for the `content` property value within).